### ISSUE TITLE
Program not running on its own (without manually running it) and not working at all for Amazon gift card reloads.

### ISSUE THREAD

**Issue by Tanman897** [jakehilborn/debbit#78]
I configured this program last month according to your examples and tutorial. I set it to run two different cards for Amazon gift card reload, each card separated into different blocks of days during the month for their transactions.

Today was the first day of the first card that was set to transact but I found that the program did not auto-open on my PC. I had to manually run it myself in order for it to initiate the transactions. Is this how it is supposed to work or is it supposed to open on its own and do its thing on its own?

Also, I attempted it for Amazon and it did not work. It asked for the OTP code, which I provided, and then gave the following error details:

ERROR: 2021-12-02 13:43:35,146 etfcu_debit_card_amazon_gift_card_reload error: Traceback (most recent call last):
  File "debbit.py", line 294, in web_automation_wrapper
  File "C:\Program Files (x86)\debbit-v2.1.2-win64\program_files\merchants\amazon_gift_card_reload.py", line 122, in web_automation

Also, do I have to manually enter my OTP for Amazon every time that it runs? That seems to thwart the point of such an automated solution.

**Comment by Tanman897** [comment]
I am wondering if there is a confliction because I have the AlwaysSmile plugin activated for Firefox. Not sure if that has anything to do with the program and the OTP error since I don't ever login to my Amazon account outside the smile subdomain and that plugin is designed to eliminate that very possibility. I did see though that when the program opened the browser window to do its thing, it successfully opened a regular Amazon domain outside the Smile subdomain.

**Comment by jakehilborn** [comment]
> Today was the first day of the first card that was set to transact but I found that the program did not auto-open on my PC. I had to manually run it myself in order for it to initiate the transactions. Is this how it is supposed to work or is it supposed to open on its own and do its thing on its own?

Debbit was designed to be running on your computer at all times. It's a bit janky since it's a foreground application so you'll see it in your task tray on Windows computers. If you leave it running all the time it will automatically do the purchases beginning on the 2nd day of every month. For example, on my computer Debbit has been running uninterrupted for months.

> Also, do I have to manually enter my OTP for Amazon every time that it runs? That seems to thwart the point of such an automated solution.

Once you get the OTP entered the cookies get saved that indicate to Amazon that this web browser has been used to log into your account before. So the OTP should only be one time. If the OTP automation code is broken, then set `hide_web_browser: no` in config.txt and manually use your mouse and keyboard to get through the OTP flow. It's not ideal but as long as you get it done once you're good.

> I am wondering if there is a confliction because I have the AlwaysSmile plugin activated for Firefox

Debbit uses an independent Firefox profile with no plugins activated so anything you do with Firefox will not interfere with Debbit.

**Comment by Tanman897** [comment]
Thanks for the response. I see then; I will just download the rbtray program and minimize it to the tray to get it off of my task bar. It doesn't seem to minimize to the tray on its own; it just sticks around in the task bar.

Okay, so I ended up having to do the manual route and then it seemed to save for when I reopened the program. But it is still giving an error message:

ERROR: 2021-12-04 18:45:10,709 etfcu_debit_card_amazon_gift_card_reload error: Traceback (most recent call last):
  File "debbit.py", line 294, in web_automation_wrapper
  File "C:\Program Files (x86)\debbit-v2.1.2-win64\program_files\merchants\amazon_gift_card_reload.py", line 160, in web_automation
    driver.find_element_by_xpath("//span[contains(text(),'Continue')]").find_element_by_xpath('../..').click()
  File "selenium\webdriver\remote\webdriver.py", line 394, in find_element_by_xpath
  File "selenium\webdriver\remote\webdriver.py", line 976, in find_element
  File "selenium\webdriver\remote\webdriver.py", line 321, in execute
  File "selenium\webdriver\remote\errorhandler.py", line 242, in check_response
selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //span[contains(text(),'Continue')]

It would seem that it cannot for the continue option.

**Comment by jakehilborn** [comment]
Debbit [v2.1.3](https://github.com/jakehilborn/debbit/releases/tag/v2.1.3) has been released. I hope this fixes your continue button issue. There have been a few bug fixes so I can't say with 100% certainty. I've enhanced the error reporting emails though so I'll be able to better tell if anything is still broken in v2.1.3.

**Comment by Tanman897** [comment]
Okay, I have now updated to the latest version and also updated this file like you suggested. Mine is set up to run the 2nd - 12th days of the month for two different cards (15 transactions total). I have already manually processed them since I had not heard back earlier and needed to complete my monthly transactions. I will have to wait till next month to see if it works; fingers crossed! :) Thanks for working on this so promptly!
